File: automap/src/main/java/com/userjhansen/automap/Maps/Map.java
package com.userjhansen.automap.Maps;

import com.acmerobotics.roadrunner.Pose2d;
import com.userjhansen.automap.AutoPart;

public interface Map {
    public Pose2d getStartPosition();

    public AutoPart[] getStartParts();
}

File: automap/src/main/java/com/userjhansen/automap/Maps/Map.kt
package com.userjhansen.automap.Maps

import com.acmerobotics.roadrunner.Pose2d
import com.userjhansen.automap.AutoPart

interface Map {

    val specimenPosition2: Pose2d
    val collectPosition: Pose2d
    val startPosition: Pose2d

//    GAME SPECIFIC OPTIMISATION
    val specimenPosition: Pose2d
    val depositPosition: Pose2d

    val intakeParts: Array<Array<AutoPart>>
    val depositParts: Array<AutoPart>
    val parkParts: Array<AutoPart>
    val specimenParts: Array<AutoPart>
}